### PR TITLE
Implement support for API Gateway v1 and v2 

### LINKED ISSUE #349: Support API Gateway
Ideally, we'd support both versions of API Gateway: 1 and 2.

### PR DESCRIPTION
**Pull request by zackproser**
## Description

Fixes #349.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

Implement support for inspecting and nuking API Gateways (v1 and v2).

### CODE REVIEW

File: aws/apigatewayv2_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	}
+
+	param := &apigatewayv2.CreateApiInput{
+		Name:         aws.String(name),
+		ProtocolType: aws.String("HTTP"),
+	}
+
+	output, err := svc.CreateApi(param)
+	if err != nil {
+		assert.Failf(t, "Could not create test API Gateway: %s", errors.WithStackTrace(err).Error())
```
Comment (zackproser) [inline aws/apigatewayv2_test.go:37]:
```suggestion
		assert.Failf(t, "Could not create test API Gateway (v2): %s", errors.WithStackTrace(err).Error())
```

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ The currently supported functionality includes:
 
 - Inspecting and deleting all ACM Private CA in an AWS account
 - Inspecting and deleting all Auto scaling groups in an AWS account
-- Inspecting and deleting all Elastic Load Balancers (Classic and V in an AWS account
+- Inspecting and deleting all Elastic Load Balancers (v1 and v2) in an AWS account
```
Comment (rhoboat) [inline README.md:15]: Eep, can't believe we had that error...

File: config/config_test.go [hunk 3]
```diff
@@ -37,6 +37,7 @@ func emptyConfig() *Config {
 		ResourceType{FilterRule{}, FilterRule{}},
 		ResourceType{FilterRule{}, FilterRule{}},
 		ResourceType{FilterRule{}, FilterRule{}},
+		ResourceType{FilterRule{}, FilterRule{}},
```
Comment (rhoboat) [inline config/config_test.go:40]: If we aren't going to ensure that the config file can be used for API Gateway (with tests), is it worth adding it to the Config?

**Review by rhoboat — COMMENTED** [review]
LGTM, except for one thing, I think it would be good to have one or two mocks for what a config yaml file would look like with API Gateway, and a test for that mock.

File: config/config_test.go (hunk 3 above)
Comment (zackproser) [reply config/config_test.go:40]: Probably not! Added Config tests for both versions of API Gateway in [695ac8f](https://github.com/gruntwork-io/cloud-nuke/pull/352/commits/695ac8fd3835b72299fb1917e08c9e6f986663bc)

**Comment by zackproser** [comment]
> LGTM, except for one thing, I think it would be good to have one or two mocks for what a config yaml file would look like with API Gateway, and a test for that mock.

Thanks for review! I've introduced complete mock tests for configuration with APIGateway v1 and v2 in [695ac8f](https://github.com/gruntwork-io/cloud-nuke/pull/352/commits/695ac8fd3835b72299fb1917e08c9e6f986663bc)

File: config/config_test.go (hunk 3 above)
Comment (rhoboat) [reply config/config_test.go:40]: Nicely done. 💯

**Review by rhoboat — APPROVED** [review]
LGTM! Very nice.

**Comment by zackproser** [comment]
Thanks for reviews! Going to merge this in now.

I'll release it as backward incompatible given https://github.com/gruntwork-io/cloud-nuke#choosing-a-new-release-tag

**Pull request merged**